app/models/metric/purging.rb

```diff
@@ -85,9 +87,32 @@ def self.purge_hourly(older_than, window = nil, total_limit = nil, &block)
   end
 
   def self.purge_realtime(older_than, window = nil, total_limit = nil, &block)
-    purge_by_date(older_than, "realtime", window, total_limit, &block)
+    truncate_child_tables(older_than)
   end
 
+  def self.truncate_child_tables(older_than)
+    target_hours = determine_target_hours(older_than, Time.now.utc)
+    return if target_hours.blank?
+
+    target_hours.each do |hour|
+      Metric.with_table_for_time(hour) do
+        Metric.truncate
+      end
+    end
+  end
+
+  def self.determine_target_hours(older_than, end_date = Time.now)
+    return [] if (end_date - older_than) > 24.hours
```

Comment:
Does this mean we could possibly never run metrics purging if the purge timer is greater than `24.hours`?  Not here, but maybe we should fall back to the old way of doing things if the truncation strategy would be acting on no tables, just incase someone mis-configures things.

Otherwise, I don't think we have a safe guard or way of warning a user that this setting should remain under `12.hours`/`24.hours` (haven't done the math on your claim in the PR description 😉 ).

Reply:
This is exactly what I was concerned about it my comments in gitter.  I am almost thinking we should completely prevent setting a time greater than or equal to 24 hours in config.

Reply:
We can do that in the `lib/vmdb/config/validator.rb` by adding a new method, `VMDB::Config::Validator#performance`, correct?

Reply:
I had a safeguard in here. but it seemed out of place to know about the interval and the duration.

I think it is probably: `duration <= 22.hours ; duration + interval + 1.hour < 24.hours` - but it just takes longer than a day to purge each hour. So I said 12.hours to be safe.